### DISCUSSION TITLE
Converting SVG containing <textPath> text to PNG

### DISCUSSION THREAD

**Discussion by ameurenand** [ImageMagick/ImageMagick#7201] [Digital Image Processing]
I'm trying to convert this SVG to PNG:

```
<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 28.4.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Calque_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 841.89 595.28" style="enable-background:new 0 0 841.89 595.28;" xml:space="preserve">
<style type="text/css">
	.st0{fill:none;}
</style>

  <path d="M250 250 L250 0 A250,250,0,0,1,466.50635094610965,124.99999999999997  L250 250 Z" fill="#fdd" stroke="#ddd"></path>
  <defs>
    <path id="p1" d="M250 50 A200,200,0,0,1,423.2050807568877,149.99999999999997" fill="#ddd" stroke="#ddd"></path>
  </defs>
  <text style="font-size: 40px;">
    <textPath xlink:href="#p1" startOffset="50%" text-anchor="middle">Text</textPath>
  </text>
</svg>
```

I tried with the latest IM version 7.1.1.29 using this command line:
`convert.exe -quiet -background none -density 90 svg:D:\Files\SVG\textpath.svg D:\Files\SVG\textpath.png`

Unfortunately the Text is not displayed.

IM convert result:
![image](https://github.com/ImageMagick/ImageMagick/assets/9213525/c4494902-048b-4596-bd76-bdad396f8f0b)

Expected convert result:
![image](https://github.com/ImageMagick/ImageMagick/assets/9213525/629e2346-cc48-4f36-889a-3c32d55f36b7)

Looks like it comes from librsvg as Inkscape well displayed it.

**Comment by snibgo** [comment]
To use textPath, you need to use Inkscape as the delegate that rasterizes SVG files. To do this, simply install Inkscape and ensure it is on the system path.

**Comment by ameurenand** [comment]
Thank you but as I use ImageMagick as library in my app, I can't use any delegate.

**Reply by fmw42** [reply]
It is not a delegate, but a separate app.  You do not have to compile Imagemagick with it.  Once installed on your system, Imagemagick should find it and use it over any internal or delegate SVG renderer.

**Reply by ameurenand** [reply]
Yes but I have to install InkScape on the customer's system.
